tests/helpers.py

```diff
@@ -0,0 +1,32 @@
+import tracemalloc
```

Comment:
Let's put this in the packaged files (under `src/stcal`)? That way it can be used from jwst, etc.

Reply:
Is this expected to be used by `STCAL` or `JWST` or just for testing?  If it's just for testing, I think it should be here.

Reply:
The expectation is that this will be used in the jwst tests. Here's one example that can be replaced with the class added here:
https://github.com/spacetelescope/jwst/blob/f3df1e805c126ed7593a4bd2447b124935a6dfa7/jwst/regtest/test_miri_image.py#L126
The overall goal is to add memory checks to stcal, jwst, romancal code to make sure future code changes don't unexpectedly increase memory usage.

Reply:
I think we're deciding that.  I think it would be good to have it available to jwst and perhaps romancal for testing, in which case it needs to be importable and therefore somewhere within `src/stcal` as Brett said.  If the `assert` statement were changed to an exception, then it could be used elsewhere, but currently I don't think there's an existing use-case for that.

https://github.com/spacetelescope/jwst/issues/8775 comes to mind - I could imagine a scenario where a custom exception would be useful if something tries to allocate more memory than allowed.  But for this particular case I still don't think this would be used, since we would want to figure out the predicted memort allocation _before_ attempting to allocate it